pulsar-client/src/main/java/org/apache/pulsar/client/impl/DefaultCryptoKeyReader.java

```diff
@@ -93,12 +93,16 @@ public EncryptionKeyInfo getPrivateKey(String keyName, Map<String, String> metad
     private byte[] loadKey(String keyUrl) throws IOException, IllegalAccessException, InstantiationException {
         try {
             URLConnection urlConnection = new URL(keyUrl).openConnection();
-            String protocol = urlConnection.getURL().getProtocol();
-            if ("data".equals(protocol) && !APPLICATION_X_PEM_FILE.equals(urlConnection.getContentType())) {
-                throw new IllegalArgumentException(
-                        "Unsupported media type or encoding format: " + urlConnection.getContentType());
+            try {
+                String protocol = urlConnection.getURL().getProtocol();
+                if ("data".equals(protocol) && !APPLICATION_X_PEM_FILE.equals(urlConnection.getContentType())) {
+                    throw new IllegalArgumentException(
+                            "Unsupported media type or encoding format: " + urlConnection.getContentType());
+                }
+                return IOUtils.toByteArray(urlConnection);
+            } finally {
+                IOUtils.close(urlConnection);
```

Comment:
we can use try-with grammar ?

Reply:
no, since there's special logic in IOUtils.close for URLConnection https://github.com/apache/commons-io/blob/fa59009aaabcf8671a8d741993ef355f42b95ccd/src/main/java/org/apache/commons/io/IOUtils.java#L354-L358